src/test/java/sdmExam/BoardShould.java

```diff
@@ -46,34 +40,40 @@ public void intersectionOutsideBoard(Position position) {
     }
 
     @ParameterizedTest
-    @MethodSource({"provideIntersectionForMarkCorrectlyAnIntersection"})
-    public void markCorrectlyAnIntersection(Intersection intersection, Stone stone) throws NoSuchElementException {
-        board.addStoneAt(Stone.BLACK, in(5, 7));
-        board.addStoneAt(Stone.WHITE, in(4, 3));
-        board.addStoneAt(Stone.WHITE, in(9, 6));
-        assertEquals(intersection.getStone(), stone);
+    @MethodSource({"providePositionForIntersectionOutsideBoard"})
+    public void intersectionOutsideBoard(Position position) {
+        assertThrows(Exception.class, () -> board.intersectionAt(position));
     }
 
-    private static Stream<Arguments> provideIntersectionForMarkCorrectlyAnIntersection() {
+    private static Stream<Arguments> provideIntersectionToMarkCorrectlyAnIntersection() {
         return Stream.of(
                 Arguments.of(board.intersectionAt(in(5, 7)), Stone.BLACK),
                 Arguments.of(board.intersectionAt(in(4, 3)), Stone.WHITE),
                 Arguments.of(board.intersectionAt(in(9, 6)), Stone.WHITE)
         );
     }
 
+    @ParameterizedTest
+    @MethodSource({"provideIntersectionToMarkCorrectlyAnIntersection"})
+    public void markCorrectlyAnIntersection(Intersection intersection, Stone stone) throws NoSuchElementException {
+        board.addStoneAt(Stone.BLACK, in(5, 7));
+        board.addStoneAt(Stone.WHITE, in(4, 3));
+        board.addStoneAt(Stone.WHITE, in(9, 6));
+        assertEquals(intersection.getStone(), stone);
+    }
 
     @TestFactory
     Stream<DynamicTest> checkOrthogonalAdjacent() {
-        board.addStoneAt(Stone.WHITE, in(7, 9));
-        board.addStoneAt(Stone.WHITE, in(3, 4));
-        board.addStoneAt(Stone.WHITE, in(4, 4));
-        board.addStoneAt(Stone.WHITE, in(12, 5));
-        board.addStoneAt(Stone.WHITE, in(13, 5));
+        Board customBoard = Board.buildTestBoard(13);
+        customBoard.addStoneAt(Stone.WHITE, in(7, 9));
+        customBoard.addStoneAt(Stone.WHITE, in(3, 4));
+        customBoard.addStoneAt(Stone.WHITE, in(4, 4));
+        customBoard.addStoneAt(Stone.WHITE, in(12, 5));
+        customBoard.addStoneAt(Stone.WHITE, in(13, 5));
 
-        Intersection firstIntersection = board.intersectionAt(in(7, 9));
-        Intersection secondIntersection = board.intersectionAt(in(3, 4));
-        Intersection thirdIntersection = board.intersectionAt(in(12, 5));
+        Intersection firstIntersection = customBoard.intersectionAt(in(7, 9));
+        Intersection secondIntersection = customBoard.intersectionAt(in(3, 4));
+        Intersection thirdIntersection = customBoard.intersectionAt(in(12, 5));
 
         List<Intersection> inputList = Arrays.asList(firstIntersection, secondIntersection, thirdIntersection);
```

Comment:
Here, and in other parts of code, instead of using Arrays.asList (which produce a mutable list) use List.of. It's more correct as we do not need to modify the state of those two lists.

Reply:
Done